eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -2526,4 +2526,22 @@ public function count(Filter $filter, ?array $languages = null): int
 
         return $this->contentFilteringHandler->count($filter);
     }
+
+    public function loadVersionInfoListByContentInfo(array $contentInfoList): array
+    {
+        $contentIds = array_column($contentInfoList, 'id');
```

Comment:
The main issue I have here is that there's no validation of user input, which is error-prone. I managed by accident to pass `VersionInfo` instead of `ContentInfo` and get some, but incorrect, results.

What I'd do here is more validation. Maybe there's better solution for that @ezsystems/php-dev-team? I was tempted to use `Webmozart\Assert\Assert::allIsInstanceOf`, but I actually prefer to still get API `InvalidArgumentException`, hence:

```php
foreach ($contentInfoList as $idx => $contentInfo) {
    if (!$contentInfo instanceof ContentInfo) {
        throw new InvalidArgumentException(
            '$contentInfoList',
            sprintf(
                'Element at position %d is not an instance of %s',
                $idx,
                $contentInfo
            )
        );
    }
}
```